package.json

```diff
@@ -10,19 +10,22 @@
     "client",
     "API"
   ],
-  "main": "lib/index.js",
+  "main": "./lib/index.js",
+  "types": "./lib/index.d.ts",
   "license": "0BSD",
   "dependencies": {
     "@daml/react": "1.3.0",
     "@daml/types": "1.3.0",
     "@mojotech/json-type-validation": "^3.1.0",
     "@types/jsonwebtoken": "^8.5.0",
     "jsonwebtoken": "^8.5.1",
+    "typescript": "~3.8.3",
     "react": "^16.12.0"
   },
   "scripts": {
     "build": "tsc --build",
     "build:watch": "tsc --build --watch",
+    "postinstall": "tsc --build",
     "test": "jest",
```

Comment:
Will this be run even if someone were to install from `npm`?

Reply:
Yep, it works with `npm` as well, just tested it

Reply:
Included a note in the README as well

Reply:
Cool, how did you test it? I haven't investigated this but can you run a local npm repo?

But moreover, for this post-install script, wouldn't we want to distribute the generated javascript instead? Or does npm take Typescript too? I could imagine that someone would want to use this library without having to ever download typescript.

Reply:
Tested by making a new directory, running `npm init`, (answer the default to all the questions), running `npm install @types/react` (needed for the post-install build to succeed) and then doing `npm install digital-asset/dabl-react#add-postinstall-hook` to add this branch. 

The npm project doesn't need to explicitly specify typescript in its package.json, or install it itself, but it does wind up in `node_modules` as a result of the dependency-of-a-dependency deal (alongside dozens of other packages that also get installed similarly).  

(Now I realize that the npm project wouldn't need to install `@types/react` either if we also move it into `dependencies` for dabl-react, like I did already for `typescript`. Currently it's a devDependency)

There's another option here which is to simply commit the `lib/` directory. Then we wouldn't even need postinstall: the generated javascript will come with the clone. I don't really like the idea of that though, as its generally not recommended to put generated files into git

Reply:
What I'm trying to understand is if this `post-install` hook gets run if we were to deploy the build artifacts to `npmjs`. Those artifacts are just compiled js, with a dependency on react, but not one on typescript. That code does not need the post install script right?